File: clients/java_graal/src/main/java/ai_cup_22/model/Order.java
package ai_cup_22.model;

import ai_cup_22.util.StreamUtil;

/**
 * Player's (team's) orders
 */
public class Order {
    /**
     * Orders for each of your units
     */
    private java.util.Map<Integer, ai_cup_22.model.UnitOrder> unitOrders;

    /**
     * Orders for each of your units
     */
    public java.util.Map<Integer, ai_cup_22.model.UnitOrder> getUnitOrders() {
        return unitOrders;
    }

    /**
     * Orders for each of your units
     */
    public void setUnitOrders(java.util.Map<Integer, ai_cup_22.model.UnitOrder> value) {
        this.unitOrders = value;
    }

    public Order(java.util.Map<Integer, ai_cup_22.model.UnitOrder> unitOrders) {
        this.unitOrders = unitOrders;
    }

    /**
     * Read Order from input stream
     */
    public static Order readFrom(java.io.InputStream stream) throws java.io.IOException {
        java.util.Map<Integer, ai_cup_22.model.UnitOrder> unitOrders;
        int unitOrdersSize = StreamUtil.readInt(stream);
        unitOrders = new java.util.HashMap<>(unitOrdersSize);
        for (int unitOrdersIndex = 0; unitOrdersIndex < unitOrdersSize; unitOrdersIndex++) {
            int unitOrdersKey;
            unitOrdersKey = StreamUtil.readInt(stream);
            ai_cup_22.model.UnitOrder unitOrdersValue;
            unitOrdersValue = ai_cup_22.model.UnitOrder.readFrom(stream);
            unitOrders.put(unitOrdersKey, unitOrdersValue);
        }
        return new Order(unitOrders);
    }

    /**
     * Write Order to output stream
     */
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeInt(stream, unitOrders.size());
        for (java.util.Map.Entry<Integer, ai_cup_22.model.UnitOrder> unitOrdersEntry : unitOrders.entrySet()) {
            int unitOrdersKey = unitOrdersEntry.getKey();
            StreamUtil.writeInt(stream, unitOrdersKey);
            ai_cup_22.model.UnitOrder unitOrdersValue = unitOrdersEntry.getValue();
            unitOrdersValue.writeTo(stream);
        }
    }

    /**
     * Get string representation of Order
     */
    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder("Order { ");
        stringBuilder.append("unitOrders: ");
        stringBuilder.append(String.valueOf(unitOrders));
        stringBuilder.append(" }");
        return stringBuilder.toString();
    }
}

File: clients/swift/Sources/AiCup22/Model/Order.swift
/// Player's (team's) orders
public struct Order {
    /// Orders for each of your units
    let unitOrders: [Int32: UnitOrder]

    /// Read Order from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> Order {
        var unitOrders: [Int32: UnitOrder]
        let unitOrdersSize = stream.readInt32()
        unitOrders = [:]
        for _ in 0..<unitOrdersSize {
            let unitOrdersKey: Int32
            let unitOrdersValue: UnitOrder
            unitOrdersKey = stream.readInt32()
            unitOrdersValue = UnitOrder.readFrom(stream)
            unitOrders[unitOrdersKey] = unitOrdersValue
        }
        return Order(unitOrders: unitOrders)
    }

    /// Write Order to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeInt32(Int32(unitOrders.count))
        for (unitOrdersKey, unitOrdersValue) in unitOrders {
            stream.writeInt32(unitOrdersKey)
            unitOrdersValue.writeTo(stream)
        }
    }
}
